### PR TITLE
Split swap forms

### PR DESCRIPTION
**Pull request by hungdoansy**
# Description
The SwapV2 page is very big now, over 1000 lines. This refactoring which was requested by @tiendv89, is the precursor to integrate the new API to encode trades.

The main goal is to slice components into smaller ones. That's why there is prop drilling over several components. I avoid making changes to the logic of those components.

The refactoring consists of the following steps:
- Create `pages/SwapV3` by duplicating `pages/SwapV2`
- Create `components/swapv3`
- Split `SwapForm` in `pages/SwapV3/index.tsx`
- Split `SwapForm` into smaller components such as: InputCurrencyPanel, OutputCurrencyPanel, PriceImpactNote, TradeComparisionNote, TradePrice and ActionButton (to swap)

### CODE REVIEW

File: src/pages/SwapV3/index.tsx [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+        [Field.INPUT]: parsedAmount,
+        [Field.OUTPUT]: parsedAmount,
+      }
+    : {
+        [Field.INPUT]: independentField === Field.INPUT ? parsedAmount : trade?.inputAmount,
+        [Field.OUTPUT]: independentField === Field.OUTPUT ? parsedAmount : trade?.outputAmount,
+      }
+
+  // reset recipient
+  useEffect(() => {
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV3/index.tsx:240]: redudant

File: src/pages/SwapV3/index.tsx [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+    onChangeRecipient(null)
+  }, [onChangeRecipient, isExpertMode])
+
+  useEffect(() => {
+    // Save current trade to store
+    onChangeTrade(trade)
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV3/index.tsx:246]: redudant

File: src/pages/SwapV3/index.tsx [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+  }
+
+  const handleRotateClick = useCallback(() => {
+    onSwitchTokensV2()
+    setIsSelectCurrencyManually(true)
+  }, [onSwitchTokensV2])
+
+  const tradeLoadedRef = useRef(0)
+  useEffect(() => {
+    tradeLoadedRef.current = Date.now()
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV3/index.tsx:294]: redudant

File: src/pages/SwapV3/index.tsx [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+      if (address === addressIn || !currencyIn) {
+        onResetSelectCurrency(Field.INPUT)
+      }
+      if (address === addressOut || !currencyOut) {
+        onResetSelectCurrency(Field.OUTPUT)
+      }
+    })
+  }, [tokenImports, chainId, prevTokenImports, currencyIn, currencyOut, onResetSelectCurrency])
+
+  useSyncTokenSymbolToUrl(currencyIn, currencyOut, onSelectSuggestedPair, isSelectCurrencyManually, isLimitPage)
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV3/index.tsx:338]: redudant

File: src/pages/SwapV3/index.tsx [hunk 5]
```diff
@@ -0,0 +1,593 @@
...
+  }, [onSwitchTokensV2])
+
+  const tradeLoadedRef = useRef(0)
+  useEffect(() => {
+    tradeLoadedRef.current = Date.now()
+  }, [trade])
+
+  const { mixpanelHandler } = useMixpanel(trade, currencies)
+
+  const onSelectSuggestedPair = useCallback(
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV3/index.tsx:299]: pass onSelectSuggestedPair to SwapForm as props. currently duplicate

File: src/pages/SwapV3/index.tsx [hunk 6]
```diff
@@ -0,0 +1,593 @@
...
+              )}
+              {activeTab === TAB.LIMIT && (
+                <LimitOrder
+                  isSelectCurrencyManual={isSelectCurrencyManually}
+                  setIsSelectCurrencyManual={setIsSelectCurrencyManually}
+                  refreshListOrder={refreshListOrder}
+                />
+              )}
+            </AppBodyWrapped>
+            {isSwapPage && <AdvancedSwapDetailsDropdown trade={trade} feeConfig={feeConfig} />}
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV3/index.tsx:532]: move to swapform

File: src/components/swapv3/ActionButton.tsx [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+    if (isSolanaUnwrap) onUserInput(Field.INPUT, balanceIn?.toExact() ?? '')
+  }, [balanceIn, isSolanaUnwrap, onUserInput, parsedAmount])
+
+  const showWrap: boolean = wrapType !== WrapType.NOT_APPLICABLE
+  const trade = showWrap ? undefined : v2Trade
+  const isPriceImpactInvalid = !!trade?.priceImpact && trade?.priceImpact === -1
+  const isPriceImpactHigh = !!trade?.priceImpact && trade?.priceImpact > 5
+  const isPriceImpactVeryHigh = !!trade?.priceImpact && trade?.priceImpact > 15
+
+  const parsedAmounts = showWrap
```
Comment (nguyenhoaidanh) [inline src/components/swapv3/ActionButton.tsx:109]: sound duplicate with SwapV3/index.ts
I think we need write utils function

File: src/pages/SwapV3/index.tsx (hunk 1 above)
Comment (hungdoansy) [reply src/pages/SwapV3/index.tsx]: I removed

File: src/pages/SwapV3/index.tsx (hunk 3 above)
Comment (hungdoansy) [reply src/pages/SwapV3/index.tsx]: I removed

File: src/pages/SwapV3/index.tsx (hunk 4 above)
Comment (hungdoansy) [reply src/pages/SwapV3/index.tsx]: I removed

File: src/pages/SwapV3/index.tsx (hunk 2 above)
Comment (hungdoansy) [reply src/pages/SwapV3/index.tsx]: I removed

File: src/pages/SwapV3/index.tsx (hunk 6 above)
Comment (hungdoansy) [reply src/pages/SwapV3/index.tsx]: Sure

**Pull request closed without merging**